fractions/jose/src/main/java/org/wildfly/swarm/jose/JoseConfiguration.java

```diff
@@ -0,0 +1,260 @@
+
+    public String keystorePath() {
+        return this.keystorePath.get();
+    }
+
+    public JoseConfiguration signatureKeyAlias(String keyAlias) {
+        this.signatureKeyAlias.set(keyAlias);
+        return this;
+    }
+
+    public String signatureKeyAlias() {
+        return this.signatureKeyAlias.get();
+    }
+
+    public JoseConfiguration signatureAlgorithm(String algorithm) {
+        signatureAlgorithm.set(algorithm);
+        return this;
+    }
+
+    public String signatureAlgorithm() {
+        return this.signatureAlgorithm.get();
+    }
+
+    public JoseConfiguration signatureFormat(JoseFormat format) {
+        signatureFormat.set(format.name());
+        return this;
+    }
+
+    public JoseConfiguration signatureDataEncoding(boolean encoding) {
+        signatureDataEncoding.set(encoding);
+        return this;
+    }
+
+    public boolean signatureDataEncoding() {
+        return this.signatureDataEncoding.get();
+    }
+
+    public JoseConfiguration signatureDataDetached(boolean detached) {
+        signatureDataDetached.set(detached);
+        return this;
+    }
+
+    public boolean signatureDataDetached() {
+        return this.signatureDataDetached.get();
+    }
+
+    public JoseFormat signatureFormat() {
+        return JoseFormat.valueOf(this.signatureFormat.get());
+    }
+
+    public JoseConfiguration encryptionFormat(JoseFormat format) {
+        encryptionFormat.set(format.name());
+        return this;
+    }
+
+    public JoseFormat encryptionFormat() {
+        return JoseFormat.valueOf(this.encryptionFormat.get());
+    }
+
+    public JoseConfiguration encryptionKeyAlias(String keyAlias) {
+        this.encryptionKeyAlias.set(keyAlias);
+        return this;
+    }
+
+    public String encryptionKeyAlias() {
+        return this.encryptionKeyAlias.get();
+    }
+
+    public JoseConfiguration encryptionKeyPassword(String password) {
+        this.encryptionKeyPassword.set(password);
+        return this;
+    }
+
+    public String encryptionKeyPassword() {
+        return this.encryptionKeyPassword.get();
+    }
+
+    public JoseConfiguration keyEncryptionAlgorithm(String algorithm) {
+        keyEncryptionAlgorithm.set(algorithm);
+        return this;
+    }
+
+    public String keyEncryptionAlgorithm() {
+        return this.keyEncryptionAlgorithm.get();
+    }
+
+    public JoseConfiguration contentEncryptionAlgorithm(String algorithm) {
+        contentEncryptionAlgorithm.set(algorithm);
+        return this;
+    }
+
+    public String contentEncryptionAlgorithm() {
+        return this.contentEncryptionAlgorithm.get();
+    }
+    /**
+     * Keystore type.
+     */
+    @Configurable("swarm.jose.keystore.type")
+    @AttributeDocumentation("Keystore type")
+    private Defaultable<String> keystoreType = string(DEFAULT_KEYSTORE_TYPE);
+
+    /**
+     * Path to the keystore.
+     */
+    @Configurable("swarm.jose.keystore.path")
+    @AttributeDocumentation("Path to the keystore")
```

Comment:
Is it only a filesystem path, or can I use a classpath resource too?

Reply:
Right now it is only classpath; I haven't looked yet at supporting the external locations. I'm not sure how to handle it. I'd like to introduce a resolver interface at some point such that it works not only for say HTTPS URLs but also also custom mediums (hardware, etc).

Reply:
This is not a feature request, just a request for clarifying the doc text :-)

Reply:
I see, let me do it quickly